caikit/runtime/service_generation/protoable.py

```diff
@@ -124,25 +124,33 @@ def is_protoable_type(arg_type: Type) -> bool:
     Or if it's an imported Caikit data model class.
     Or if it's a Union of at least one of those.
     Or if it's a List of one of those.
+    Or if it's a Dict of one of those.
     False otherwise"""
     proto_primitive_set = list(DATAOBJECT_PY_TO_PROTO_TYPES.keys())
-
+    protoable = False
     if arg_type in proto_primitive_set:
-        return True
-    if is_data_model_type(arg_type):
-        return True
-
-    if typing.get_origin(arg_type) == list:
+        protoable = True
+    elif is_data_model_type(arg_type):
+        protoable = True
+    elif typing.get_origin(arg_type) == list:
         log.debug2("Arg is List")
         if len(typing.get_args(arg_type)) == 0:
             log.debug2("List annotation has no type")
             return False
-        return typing.get_args(arg_type)[0] in proto_primitive_set
-
-    if typing.get_origin(arg_type) == Union:
+        protoable = typing.get_args(arg_type)[0] in proto_primitive_set
+    elif typing.get_origin(arg_type) == dict:
+        log.debug2("Arg is Dict")
+        if len(typing.get_args(arg_type)) == 0:
+            log.debug2("Dict annotation has no type")
+            return False
```

Comment:
nit: this is mixing the "return early" pattern with the "save and return later" pattern. I don't really mind picking one over the other as long as we're consistent